src/styles/calendar.scss

```
@@ -0,0 +1,54 @@
+$size: 100px;
+
+$margin: (
+  mon: 0,
+  tue: 101px,
+  wed: 202px,
+  thu: 303px,
+  fri: 404px,
+  sat: 505px,
+  sun: 606px,
+);
+
+.calendar {
+  display: flex;
+  flex-wrap: wrap;
+  width: 7 * $size + 7px;
+  margin: 50px auto;
+
+  div:nth-child(n) {
+    @include flex(flex,center,center);
+    width: $size;
+    height: $size;
+    border: 1px solid black;
+    margin: 0 1px 1px 0;
+    background-color: #eee;
+    transition: background-color 0.5s, transform 0.5s;
+
+    &:hover {
+      background-color: #ffbfcb;
+      cursor: pointer;
+      transform: translate(0, -20px);
+    }
+  }
+
+  @for $i from 1 through 31 {
+    div:nth-child(#{$i})::before {
+      content: "#{$i}";
+    }
+  }
+
+  @for $i from 28 through 30 {
+    &-28 div:nth-child(#{$i + 1}),
+    &-29 div:nth-child(#{$i + 2}),
+    &-30 div:nth-child(#{$i + 3}) {
+      display: none;
```

За ці рядки я вже писав що можна ще більше зменшити бо в такому виді не бачу сенсу цикл використовувати. Хоча не знаю на скільки саме моя формула для nth-child  підходить у твоїй ситуації, але мало б правильно обирати..
```suggestion
  @for $i from 28 through 30 {
    &-#{$i} div:nth-child(n + #{$i + 1}) {
      display: none;
```

---

Погоджуюся з Юрою.